### PR TITLE
21.7 fb hansen queryto file xml

### PR DESCRIPTION
**Pull request by jonesgaohsu**
#### Rationale
Per Request added the schema for StudyDetails_RandalData to onprc_ehr.xml

#### Related Pull Requests
* 

#### Changes
*

### CODE REVIEW

**Comment by jonesgaohsu** [comment]
JOsh requested your review based on your request.  Thanks for keeping an eye on our code

**Comment by labkey-jeckels** [comment]
@jonesgaohsu TeamCity is still reporting a mismatch - DPCO vs dpc0 (the letter O vs zero). Are you checking the test results that are linked from the pull request?

Additionally, the new queries are broken on TeamCity because it uses a different folder structure than PRIMe does. Instead of using "/ONPRC/EHR", please use "Site.{substitutePath moduleProperty('EHR','EHRStudyContainer')}" which will resolve the appropriate container path for the server on both TeamCity and PRIMe.

**Comment by jonesgaohsu** [comment]
Thanks I will make the changes

**Comment by jonesgaohsu** [comment]
Thanks and It appears I am not using the results correctly,  will step back and review the correct process and make sure I understand the process and results

**Comment by labkey-jeckels** [comment]
@jonesgaohsu do you expect to work on this more today? The test failures are causing noise in other branches and repositories since they've been merged to the main 21.7 branch already, so I'd like to get them resolved ASAP.

**Comment by jonesgaohsu** [comment]
completed update to correct left over " from query and verified thru dev instance

File: onprc_ehr/resources/etls/HansenQueriesToFile.xml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 <?xml version="1.0" encoding="UTF-8" ?>
 <etl xmlns="http://labkey.org/etl/xml">
-    <name>mfshToRandalData</name>
+21.7_fb_HansenQuerytoFileXML
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/etls/HansenQueriesToFile.xml:3]: This looks like a copy/paste problem

File: onprc_ehr/resources/schemas/onprc_ehr.xml [hunk 2]
```diff
@@ -1306,4 +1308,33 @@
...
+            <column columnName="grp" />
+            <column columnName="grp_order" />
+            <column columnName="grp_id" />
+            <column columnName="rhCode" />
+            <column columnName="grpnm" />
+            <column columnName="sex" />
+            <column columnName="cohortStart" />
+            <column columnName="cohortend" />
+            <column columnName="Do" />
+            <column columnName="DPCO" />
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/schemas/onprc_ehr.xml]: This should be DPC0, not DPCO

```suggestion
            <column columnName="DPC0" />
```

File: onprc_ehr/resources/queries/onprc_ehr/RandData_Birth.sql [hunk 3]
```diff
@@ -0,0 +1,32 @@
+SELECT d.Id,
+       d.date,
+       d.date_type,
+       d.birth_condition,
+       d.room,
+       d.cge,
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/queries/onprc_ehr/RandData_Birth.sql]: This is causing a test failure. I assume it should be cage.

```suggestion
       d.cage,
```

File: onprc_ehr/resources/queries/onprc_ehr/RandData_Assignment.sql [hunk 4]
```diff
@@ -0,0 +1,21 @@
+SELECT d.Id,
+        d.project,
+        d.date,
+        d.projectedRelease,
+        d.enddate,
+        d.dType,
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/queries/onprc_ehr/RandData_Assignment.sql]: There is no dType column on the assignment dataset. Please correct or remove.

File: onprc_ehr/resources/queries/onprc_ehr/RandData_MHCData.sql [hunk 5]
```diff
@@ -0,0 +1,11 @@
+SELECT d.rowid,
+       d.subjectId,
+       d.datatype,
+       d.marker,
+       d.result,
+       d.score,
+       d.assaytype,
+       d.container,
+       d.totalTests
+ FROM  StudyDetails_RandalData s, Site.{substitutePath moduleProperty('EHR','EHRStudyContainer')}.study.mhcdata d
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/queries/onprc_ehr/RandData_MHCData.sql]: There is no mhcdata table or query in the study schema. Please remove or correct.

File: onprc_ehr/resources/queries/onprc_ehr/RandData_Flags.sql [hunk 6]
```diff
@@ -0,0 +1,21 @@
+SELECT d.Id,
+       d.date,
+       d.enddate,
+       d.flag,
+       d.performedby,
+       d.remark,
+       d.taskid,
+       d.requestid,
+       d.description,
+       d.RequestedBy,
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/queries/onprc_ehr/RandData_Flags.sql:10]: This column isn't part of the dataset definition that the test is loading so it's breaking. I'll add it and we can review the change in the Tuesday call.

File: onprc_ehr/resources/queries/onprc_ehr/RandData_AnimalGroups.sql [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+       d.qcstate,
+       d.remark,
+       d.performedby,
+       d.taskid,
+       d.requestid,
+       d.description,
+       d.Container,
+       d.isActive,
+       d.history,
+       d.isAssignedAtTime,
```
Comment (labkey-jeckels) [inline onprc_ehr/resources/queries/onprc_ehr/RandData_AnimalGroups.sql:15]: The test's definitions for the animal_group_members table doesn't include a project field, which means that ONPRC_EHRCustomizer isn't injecting the isAssignedAtTime and isAssignedToProtocolAtTime calculated fields. I'll add that field too.

**Review by labkey-jeckels — CHANGES_REQUESTED** [review]

**Comment by labkey-jeckels** [comment]
Also, this pull request is currently targeting develop, not the 21.7 branch.

File: onprc_ehr/resources/queries/onprc_ehr/RandData_Assignment.sql (hunk 4 above)
Comment (jonesgaohsu) [reply onprc_ehr/resources/queries/onprc_ehr/RandData_Assignment.sql]: CHanges have been made and successfully loaded

File: onprc_ehr/resources/schemas/onprc_ehr.xml (hunk 2 above)
Comment (jonesgaohsu) [reply onprc_ehr/resources/schemas/onprc_ehr.xml]: Changes made and successful passed auto testing

File: onprc_ehr/resources/queries/onprc_ehr/RandData_Assignment.sql (hunk 4 above)
Comment (jonesgaohsu) [reply onprc_ehr/resources/queries/onprc_ehr/RandData_Assignment.sql]: This should have been assignment type and was reloaded this morning to correct the issue

**Review by labkey-jeckels — APPROVED** [review]
Let's go ahead and merge this. There are still two queries that are failing validation during the tests' study import (onprc_ehr.RandData_Flags and onprc_ehr.RandData_MHCData) but it's an improvement over the status quo. We'll need to do another followup FB to address them.

**Comment by jonesgaohsu** [comment]
Merging to production instance, this is isoltated to the Hansen Lab

**Pull request merged**